core/src/oidc_auth.rs

```diff
@@ -41,9 +42,38 @@ pub enum Error {
     NoIdpServerInfo,
     CsrfMismatch,
     HumanFlowUnsupported,
+    Timedout,
     Other(String),
 }
 
+impl From<Error> for mongodb::error::Error {
+    fn from(e: Error) -> Self {
+        mongodb::error::Error::custom(e)
+    }
+}
+
+pub async fn oidc_call_back(params: CallbackContext) -> mongodb::error::Result<IdpServerResponse> {
```

Comment:
Please, here too add a comment explaining that depending on the info we need to re-auth "from scratch" or we can use the refresh token when provided.